### DISCUSSION TITLE
Wiki section for debugging

### DISCUSSION THREAD

**Discussion by nekopep** [FoobarOy/foomuuri#45] [Ideas]
Hello,

As a basic user I hit some issues while starting to use foomuuri.
Here are my feedback for today session mostly related to kde nm-applet not configuring foomurri zone for selected interface.

So, I had a good foomuri configuration file with simple zones.
I set the zone in KDE network manager:
![image](https://github.com/user-attachments/assets/e718f441-0b69-4263-a3bb-e9186e10306b)

Still restarting the connection, the firewall was always blocking all connecions :/

First: follow (-f) kernel logs (-k) in terminal: 
#journalctl -f -k

If there is no zone active on your interface REJECT and DROP message will look like this:
![image](https://github.com/user-attachments/assets/1577d98b-0a29-40b0-829d-ed34619745d7)

If zone is active:
![image](https://github.com/user-attachments/assets/2e1a0e47-7e60-4220-93e5-eea6be73ca20)

If kde network-manager seems to have no effect check foomuuri-firewalld emulation packet installation:
#apt install apt install foomuuri-firewalld
Perhaps needs a restart.

Another way to check if it working in realtime:
#foomuuri dbus 
 D-Bus handler ready
Interface "wlan0" change to zone "home"
Interface "wlan0" remove from zone ""
Interface "wlan0" change to zone "home"

Will help checking in realtime interface assignation.

Remaining question: Any option to get currently interface and their assigned interface?
like "#foomuuri zones"?

With firewalld it seem you can do "ip a show dev wlan0" and
3: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 8c:3b:4a:45:57:ac brd ff:ff:ff:ff:ff:ff
    altname wlp2s0
    inet 192.168.1.89/24 brd 192.168.1.255 scope global dynamic noprefixroute wlan0

And the Zone shoudl be displayed after scope global dynamic.
I found no way with foomuuri other than the two methods presented above.

**Comment by nekopep** [comment]
Check service status ...
$ systemctl status foomuuri
○ foomuuri.service - Multizone bidirectional nftables firewall
     Loaded: loaded (/lib/systemd/system/foomuuri.service; **disabled**; preset: enabled)
     Active: **inactive (dead)**
       Docs: https://github.com/FoobarOy/foomuuri/wiki

// Always start foomuuri on boot
#systemctl enable foomuuri
// Start it manually
#systemctl start foomuuri

**Comment by kimheino** [comment]
> With firewalld it seem you can do "ip a show dev wlan0" and 3: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000 link/ether 8c:3b:4a:45:57:ac brd ff:ff:ff:ff:ff:ff altname wlp2s0 inet 192.168.1.89/24 brd 192.168.1.255 scope global dynamic noprefixroute wlan0
> 
> And the Zone shoudl be displayed after scope global dynamic.

Quick question about this one. Are you sure about this? I tried on latest Fedora:

```
(16:31) b@cobra:~> sudo firewall-cmd --get-zone-of-interface=wlp0s20f3
home

(16:31) b@cobra:~> sudo ip a show dev wlp0s20f3 
...nothing about zone or home...
```

**Comment by kimheino** [comment]
Commit 41c4612 adds `status` command:

```
(11:58) b@cobra:~> sudo foomuuri status
Foomuuri is running
Foomuuri-dbus is running, PID 11334
Foomuuri-monitor is not running

Zones:
  localhost       
  public          
  home            wlp0s20f3
  kvm             virbr0
```
